diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -230,6 +230,11 @@ fun ASTNode.findAllNodesWithSpecificType(elementType: IElementType): List<ASTNod
     }
 }
 
+/**
+ * Check a node of type CLASS if it is a enum class
+ */
+fun ASTNode.isClassEnum(): Boolean = this.findChildByType(MODIFIER_LIST)?.hasChildOfType(ElementType.ENUM_KEYWORD) ?: false
```

Comment:
```suggestion
fun ASTNode.isClassEnum(): Boolean = (psi as? KtClass)?.isEnum() ?: false
```
What do you think?

Reply:
I think it much better. I didn't know about this.